### PR TITLE
Manage errors when environment cannot be read

### PR DESCRIPTION
**Pull request by denaitre**
Environment can be empty:
1. when current does not exist/is empty
2. when '--env' is provided with a wrong path

When the command is 'up' or 'deploy' it creates a new directory.
Otherwise, an error appears depending on the command.
This commit exits the process with an appropriate output in such situation.

### CODE REVIEW

File: enos/enos.py [hunk 1]
```diff
@@ -182,6 +182,11 @@ def up(provider=None, env=None, **kwargs):
 def install_os(env=None, **kwargs):
     logging.debug('phase[os]: args=%s' % kwargs)
 
+    if 'provider_net' not in env:
```
Comment (msimonin) [inline enos/enos.py]: Maybe we could have a util function that check it. This way we could unit test this part easily.
We want to check two things
* if a key exist in a dict
* or if the value associated is not None in some cases

What about having `check_env(env, present=[], notNone=[])` where `env` is the environnement to check, `present` are the keys that needs to be present and `notNone` are the key whose values can't be None. This function throws an exception where something wrong is found.

Comment (msimonin) [inline enos/enos.py]: Even better (IMO), make use of a decorator to validate the env prior to the task execution.

Comment (denaitre) [inline enos/enos.py]: :+1:

**Comment by denaitre** [comment]
@rcherrueau Done!

File: enos/utils/enostask.py [hunk 2]
```diff
@@ -81,6 +82,15 @@ def decorated(*args, **kwargs):
 
             # Constructs the environment
             kwargs['env'] = make_env(kwargs['--env'])
+
+            # Check if the environment has been loaded for the commands
+            # requiring it
+            if fn.__name__ not in ['up', 'deploy', 'info'] and \
```
Comment (msimonin) [inline enos/utils/enostask.py]: I maybe miss something but it looks like your previous proposition covered more cases (e.g validation for the install-os were made)
That's why I was proposing to have a dedicated decorator whose job was to check some missing / None values in the env before launching the task. What about : 
```
@enostask("""doc""")
@validation(present=['key1', 'key2'], not_None=['key3'])
```
That would pass only if `key1` and `key2` are present and `key3` not None

Comment (rcherrueau) [inline enos/utils/enostask.py]: @msimonin If the idea is to produce contracts around functions, then why not leveraging the type system of Python3? 

In other case, if the idea is to ensure that the linked env is a file, I would recommend to move following lines after a `else` at line 40.

Comment (msimonin) [inline enos/utils/enostask.py]: Yes, checking if the linked env is one thing.
But I think the original idea was to validate that some specific keys were present/not None in the env for a given task. That's why I proposed a specific decorator to separate the concerns. It looks weird to me to reference the phases (`up`, `deploy`, ...) in this file.

**Comment by msimonin** [comment]
up ?

**Comment by denaitre** [comment]
@rcherrueau @msimonin I pushed a new version of the code that adds a decorator to check if the environment file exists.

**Pull request merged**